modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java

```diff
@@ -783,8 +786,39 @@ private void prepareAndExecuteOperation(CoreEvent event, Supplier<ExecutorCallba
 
     ExecutionContextAdapter<T> operationContext = oep.getExecutionContextAdapter();
 
-    executeOperation(operationContext, mapped(callbackSupplier.get(), operationContext,
-                                              isTargetWithPolicies(event) ? valueReturnDelegate : returnDelegate));
+    setCurrentEvent((PrivilegedEvent) event);
+    boolean wasProcessorPathSet = setCurrentLocation();
+    try {
+      executeOperation(operationContext, mapped(callbackSupplier.get(), operationContext,
+                                                isTargetWithPolicies(event) ? valueReturnDelegate : returnDelegate));
+    } finally {
+      unsetCurrentLocation(wasProcessorPathSet);
+    }
+  }
+
+  private boolean setCurrentLocation() {
+    ComponentLocation componentLocation = getLocation();
```

Comment:
don't do this on every execution. Cannot you do this on the `initialise()` and cache the result?

Reply:
No, because this method sets the location in the MDC, which is thread local. We need to set it in the thread that is going to execute the operation, and we don't know which is that thread on `initialise()`.
Probably the method should be renamed to "addComponentLocationToMDC()"

Reply:
That's not what I meant. The MDC you need to set everytime, sure. But the resolved location is fixed. What I meant is that you shouldn't be calculating the location on every single request. Makes sense?